config.go

```diff
@@ -16,6 +16,12 @@ type Config struct {
 	// Packages is the slice of Nix packages that devbox makes available in
 	// its environment.
 	Packages []string `cue:"[...string]" json:"packages"`
+
+	// Shell configures the devbox shell environment.
+	Shell struct {
```

Comment:
In terms of schema, I'm tempted to keep it flat for the time being and just make it `shell_hook` at the top-level (I might regret that later if we need more things inside shell, hehe)